### PR TITLE
Add support for http connect proxying to tls connections. Fixes #54

### LINKED ISSUE #54: Support HTTP Connect proxying for TLS connections

### PR DESCRIPTION
**Pull request by plorenz**

### CODE REVIEW

File: proxies/http_connect.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+}
+
+type HttpConnectProxyDialer struct {
+	address string
+	auth    *proxy.Auth
+	timeout time.Duration
+}
+
+func (self *HttpConnectProxyDialer) Dial(network, addr string) (net.Conn, error) {
+	c, err := net.Dial(network, self.address)
```
Comment (andrewpmartinez) [inline proxies/http_connect.go:47]: This assumes the proxy server is not TLS itself?

**Review by andrewpmartinez — APPROVED** [review]
Looks good only if we are not expecting to support (TLS) HTTPS proxies. I assume we are looking to support enterprise networks that aren't locking down an internal HTTP proxy.

File: proxies/http_connect.go (hunk 1 above)
Comment (andrewpmartinez) [reply proxies/http_connect.go:47]: One way to handle TLS vs non-TLS HTTP proxies is to require the protocol prefix on the address and inspect it:

`https://proxyhost` vs `htt://proxyhost` and then do the right thing.

Comment (plorenz) [reply proxies/http_connect.go:47]: Yeah, this is just to get the bare minimum working. If/when we get more feature requests we can look at how to expand this out

**Pull request merged**